pkg/retry/retry.go

```diff
@@ -83,18 +94,23 @@ func (s *Server) handleRetry() http.Handler {
 					continue
 				}
 
-				// redeliver the failed event
+				logger.Infow("redeliver failed event", "eventID", event.ID)
+
 				if err := s.github.RedeliverEvent(ctx, *event.ID); err != nil {
 					logger.Errorw("failed to call RedeliverEvent", "code", http.StatusInternalServerError, "body", errCallingGitHub,
 						"method", "RedeliverEvent", "error", err)
 					w.WriteHeader(http.StatusInternalServerError)
 					fmt.Fprint(w, http.StatusText(http.StatusInternalServerError))
 					return
 				}
+
+				// append to the total failure events counter
+				failedEventCount += 1
 			}
 
-			// periodic update to checkpoint
-			if err := s.datastore.WriteCheckpointID(ctx, s.checkpointTableID, "checkpointID", time.Now().String()); err != nil {
+			logger.Infow("write latest checkpoint", "checkpoint", cursor)
+
+			if err := s.datastore.WriteCheckpointID(ctx, s.checkpointTableID, cursor, time.Now().String()); err != nil {
```

Comment:
1. Please grab a copy of `time.Now()` at the very start of the invocation, otherwise you'll have races.

2. Please use `UTC` of every and all times.

3. Is `String` the correct function? I would expect `.Format()` in a specific format instead.